python/test_infra/test_utils.py

```diff
@@ -287,152 +283,78 @@ def wrapper(*args, **kwargs):
-    @compile_as_mlir_module((32, 32), (32, 32))
-    def test_add(in0: Operand, in1: Operand, builder: TTIRBuilder):
-        # CHECK: %0 = tensor.empty() : tensor<32x32xf32>
-        # CHECK: %1 = "ttir.add"(%arg0, %arg1, %0)
-        # CHECK: return %1 : tensor<32x32xf32>
-
-        return builder.add(in0, in1)
-        ```
+    Wrapper around `ttmetal_to_flatbuffer_file` pybound pass.
     """
 
-    def decorator(fn: Callable):
+    def decorator(test_fn: Callable):
         def wrapper(*args, **kwargs):
-            input_file_name = fn(*args, **kwargs)
+            # Get the TTMetal module by calling the wrapped function.
+            module = test_fn(*args, **kwargs)
 
-            assert isinstance(input_file_name, str) and os.path.isfile(
-                input_file_name
-            ), (
+            assert isinstance(module, Module), (
                 f"Make sure `ttir_to_ttmetal` which was decorated with this function "
-                f"returns file name, not module."
-            )
-
-            res = _run_ttmlir_translate_ttmetal(input_file_name, output_file_name)
-
-            print(
-                f"Flatbuffer file for TTMetalBinary {output_file_name} successfully generated."
-            )
-
-            return res.returncode
-
-        return wrapper
-
-    return decorator
-
-
-def translate_ttnn_to_flatbuffer(output_file_name: str = "ttnn_fb.ttnn"):
-    """
-
-    By running `ttmlir-translate` on input file, it produces TTNN flatbuffer file
-    `output_file_name`, meant to be used as a decorator on top of `ttir_to_ttnn`
-    decorator.
-
-    Wrapper around `ttmlir-translate` call.
-
-    Example
-    -------
-
-    ```python
-    @translate_ttnn_to_flatbuffer(output_file_name="ttnn_fb_test_add.ttm")
-    @ttir_to_ttnn(dump_to_file=True, output_file_name="test_add.mlir")
-    @compile_as_mlir_module((32, 32), (32, 32))
-    def test_add(in0: Operand, in1: Operand, builder: TTIRBuilder):
-        # CHECK: %0 = tensor.empty() : tensor<32x32xf32>
-        # CHECK: %1 = "ttir.add"(%arg0, %arg1, %0)
-        # CHECK: return %1 : tensor<32x32xf32>
-
-        return builder.add(in0, in1)
-        ```
-    """
-
-    def decorator(fn: Callable):
-        def wrapper(*args, **kwargs):
-            input_file_name = fn(*args, **kwargs)
-            assert isinstance(input_file_name, str) and os.path.isfile(
-                input_file_name
-            ), (
-                f"Make sure `ttir_to_ttnn` which was decorated with this function "
-                f"returns file name, not module."
+                f"returns module, not file name."
             )
 
-            res = _run_ttmlir_translate_ttnn(input_file_name, output_file_name)
-
-            print(
-                f"Flatbuffer file for TTNNBinary {output_file_name} successfully generated."
-            )
+            # Convert to flatbuffer file.
+            ttmetal_to_flatbuffer_file(module, output_file_name, golden_info)
```

Comment:
Which `golden_info`?

Reply:
fixed. I mistakenly removed a golden dict for metal path (ttnn path still has it).